hook.php

```diff
@@ -95,23 +96,55 @@ function plugin_mreporting_install()
    ) ENGINE=InnoDB DEFAULT CHARSET={$default_charset} COLLATE={$default_collation} ROW_FORMAT=DYNAMIC;";
 
     // add display preferences
-    $query_display_pref = "SELECT id
-      FROM glpi_displaypreferences
-      WHERE itemtype = 'PluginMreportingConfig'";
-    $res_display_pref = $DB->query($query_display_pref);
-    if ($DB->numrows($res_display_pref) == 0) {
-        $queries[] = "INSERT INTO `glpi_displaypreferences`
-         VALUES (NULL,'PluginMreportingConfig','2','2','0');";
-        $queries[] = "INSERT INTO `glpi_displaypreferences`
-         VALUES (NULL,'PluginMreportingConfig','3','3','0');";
-        $queries[] = "INSERT INTO `glpi_displaypreferences`
-         VALUES (NULL,'PluginMreportingConfig','4','4','0');";
-        $queries[] = "INSERT INTO `glpi_displaypreferences`
-         VALUES (NULL,'PluginMreportingConfig','5','5','0');";
-        $queries[] = "INSERT INTO `glpi_displaypreferences`
-         VALUES (NULL,'PluginMreportingConfig','6','6','0');";
-        $queries[] = "INSERT INTO `glpi_displaypreferences`
-         VALUES (NULL,'PluginMreportingConfig','8','8','0');";
+    $query_display_pref = [
+        'SELECT' => [
+            'id',
+        ],
+        'FROM'   => DisplayPreference::getTable(),
+        'WHERE'  => [
+            'itemtype' => 'PluginMreportingConfig',
+        ],
+    ];
+
+    $res_display_pref = $DB->request($query_display_pref);
+    if ($res_display_pref->numrows() == 0) {
+        $display_preference = new DisplayPreference();
+        $display_preference->add([
+            'itemtype' => 'PluginMreportingConfig',
+            'num'    => '2',
+            'rank'     => '2',
+            'users_id' => 0,
+        ]);
+        $display_preference->add([
+            'itemtype' => 'PluginMreportingConfig',
+            'num'    => '3',
+            'rank'     => '3',
+            'users_id' => 0,
+        ]);
+        $display_preference->add([
+            'itemtype' => 'PluginMreportingConfig',
+            'num'    => '4',
+            'rank'     => '4',
+            'users_id' => 0,
+        ]);
+        $display_preference->add([
+            'itemtype' => 'PluginMreportingConfig',
+            'num'    => '5',
+            'rank'     => '5',
+            'users_id' => 0,
+        ]);
+        $display_preference->add([
+            'itemtype' => 'PluginMreportingConfig',
+            'num'    => '6',
+            'rank'     => '6',
+            'users_id' => 0,
+        ]);
+        $display_preference->add([
+            'itemtype' => 'PluginMreportingConfig',
+            'num'    => '8',
+            'rank'     => '8',
+            'users_id' => 0,
+        ]);
     }
 
     $queries[] = "CREATE TABLE IF NOT EXISTS `glpi_plugin_mreporting_notifications` (
```

Comment:
`foreach` is not necessary; it will probably be more readable to have each CREATE statement as follows:

$query = ...
$DB->doQuery($query);